### DISCUSSION TITLE
Persistent storage on the Arduino Nano Matter?

### DISCUSSION THREAD

**Discussion by technoblogy** [SiliconLabs/arduino#75] [Q&A] [answered]
Does the Arduino Nano Matter have any form of persistent storage capability? I need this for my application.

I’m thinking of one of the following possibilities:

* EEPROM. For example the ATmega328 used on the Arduino Uno provided 2 Kbytes of EEPROM, and this is provided or emulated in most subsequent Arduino boards.

* Flash storage. For example, the ATSAMD21-based boards support storage in the program Flash memory.

* LittleFS flash filing system. For example, the ESP32 and RP2040-based boards support storage in the program flash memory using the LittleFS filing system.

* DataFlash chip. For example, the Adafruit M4 and other "Express" boards include a DataFlash chip which can be used for persistent storage.

Thanks! David

**Comment by silabs-bozont** [comment]
Hello David,
EEPROM library support is coming in the next release!

**Comment by technoblogy** [comment]
Great! Look forward to trying it out.

**Comment by silabs-bozont** [comment, marked as answer]
Hello @technoblogy,
EEPROM library support is now available with 2.1.0.
See: https://github.com/SiliconLabs/arduino/tree/main/libraries/EEPROM

**Comment by technoblogy** [comment]
When will 2.1.0 be available in Arduino Boards Manager? It's still only offering me 2.0.0.

**Reply by silabs-bozont** [reply]
2.1.0 should be out by now - if the IDE doesn't offer the upgrade you can paste `https://siliconlabs.github.io/arduino/package_arduinosilabs_index.json` in the 'Additional Boards Manager URLs' field - then the update will be visible.

**Reply by technoblogy** [reply]
Thanks - I've updated and will try it out.
